### PR TITLE
remove default namespace

### LINKED ISSUE #2246: build_python_component does not work in KF 6.2
compiler.build_python_component uses the default name space for pod creation. As of KF 6.2 the notebooks are running under a user name space( ex kubeflow-<username>) with editor service account. This service account does not have access to default kubeflow name space by design. 

SDK needs to be updated to use the user service account instead. 

sample error: 

`HTTP response body: {"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"pods is forbidden: User \"system:serviceaccount:kubeflow-chavoshi:default-editor\" cannot create resource \"pods\" in API group \"\" in the namespace \"kubeflow\"","reason":"Forbidden","details":{"kind":"pods"},"code":403}
`
screen shot: 
https://screenshot.googleplex.com/QotFfe0hgnW.png

### PR DESCRIPTION
**Pull request by SinaChavoshi**
fixes #2246

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/2250)

### CODE REVIEW

File: sdk/python/kfp/containers/_component_builder.py [hunk 1]
```diff
@@ -177,7 +177,7 @@ def build_python_component(component_func, target_image, base_image=None, depend
     staging_gcs_path (str): GCS blob that can store temporary build files
     target_image (str): target image path
     timeout (int): the timeout for the image build(in secs), default is 600 seconds
-    namespace (str): the namespace within which to run the kubernetes kaniko job, default is "kubeflow"
```
Comment (gaoning777) [inline sdk/python/kfp/containers/_component_builder.py]: Would you mind adding a little more comments that the build_python_component internally will find the current namespace if it is run in the GKE cluster.

Comment (SinaChavoshi) [reply sdk/python/kfp/containers/_component_builder.py]: good point ! absolutely.

File: sdk/python/kfp/containers/_component_builder.py [hunk 2]
```diff
@@ -275,7 +275,8 @@ def build_docker_image(staging_gcs_path, target_image, dockerfile_path, timeout=
     target_image (str): gcr path to push the final image
     dockerfile_path (str): local path to the dockerfile
     timeout (int): the timeout for the image build(in secs), default is 600 seconds
-    namespace (str): the namespace within which to run the kubernetes kaniko job, default is "kubeflow"
+    namespace (str): the namespace within which to run the kubernetes kaniko job. Default is None, build_python_component
```
Comment (gaoning777) [inline sdk/python/kfp/containers/_component_builder.py]: Could you copy the comments to the build_python_component func as well? 
And, adjust the words here "build_python_component"

Comment (SinaChavoshi) [reply sdk/python/kfp/containers/_component_builder.py]: done

**Pull request merged**